### PR TITLE
Create a profile bar

### PR DESCRIPTION
**Pull request by iamlordvodemort**
- Should display user name
- Button directs to profile details screen

### CODE REVIEW

**Comment by Nigh7Sh4de** [comment]
We are not ready to merge into master yet

File: www/index.html [hunk 1]
```diff
@@ -1,73 +1,72 @@
 <!DOCTYPE html>
 <html>
 
-<head>
-    <meta charset="utf-8">
-    <meta name="viewport" content="initial-scale=1, maximum-scale=1, user-scalable=no, width=device-width">
-    <title></title>
+    <head>
```
Comment (Nigh7Sh4de) [inline www/index.html:4]: Please don't bundle code style refactors like whitespace such as this into feature pull requests

File: www/js/app.js [hunk 2]
```diff
@@ -1,10 +1,6 @@
 Stripe.setPublishableKey("pk_test_WwMFp1CE94C8P8QLtPrzW5Lq"); 
 
-var app = angular.module('starter', 
-    ['ionic', 
-     'ngCordova', 
-     'ion-datetime-picker', 
-     'credit-cards'])
+var app = angular.module('starter', ['ionic', 'ngCordova', 'ion-datetime-picker', 'credit-cards'])
```
Comment (Nigh7Sh4de) [inline www/js/app.js:3]: Please leave code style as it was. Lists of more than three items should be on separate lines for readability.

File: www/js/app.js [hunk 3]
```diff
@@ -17,6 +13,7 @@ var app = angular.module('starter',
             }
         });
     })
+
```
Comment (Nigh7Sh4de) [inline www/js/app.js]: Again, watch out for these whitespace changes, especially for dot-chained functions there should not be a new line in the chain

File: www/js/app.js [hunk 4]
```diff
@@ -55,10 +52,44 @@ var app = angular.module('starter',
...
         $urlRouterProvider.otherwise('/');
     }) 
     
+    .directive('profileBarWidget', function($ionicGesture) {
+        return {
+            restrict: 'AE',
+            // declare the directive scope as private (and empty)
+            scope: {},
+            // add behaviour to our buttons and use a variable value
+            template:   '<div class="thumbnail">'+
```
Comment (Nigh7Sh4de) [inline www/js/app.js]: Please move the template out to an HTML partial (check the `templates` folder, or consider making a new `directives` folder) and then use a `templateUrl` to reference it

File: www/js/app.js [hunk 5]
```diff
@@ -55,10 +52,44 @@ var app = angular.module('starter',
...
+                        '<div class="profile-name">'+       
+                            '<h3> {{ profile.name }} </h3>'+
+                        '</div>'+
+                        '<div class="edit-button">'+
+                            '<button class="button button-positive button-small"'+
+                               'onclick="window.location=\'#/profile\'">Edit</button>'+
+                        '</div>',
+            // we just declare what we need in the above template
+            controller: function(userInfoService, $scope) {
+                userInfoService.getProfileInfo().then(function(userInfo) {
```
Comment (Nigh7Sh4de) [inline www/js/app.js]: Controllers should be in their own file

File: www/js/settingsCtrl.js [hunk 6]
```diff
@@ -1,26 +1,17 @@
 angular.module('starter').controller("Settings", function ($scope, $stateParams, $ionicModal, $state, $ionicPopup, userInfoService) {
+  
+  $scope.rating = 42;
```
Comment (Nigh7Sh4de) [inline www/js/settingsCtrl.js]: What is the purpose of this $scope variable?

File: www/js/app.js (hunk 5 above)
Comment (Nigh7Sh4de) [reply www/js/app.js]: or, consider moving the entire directive definition into a new file in which case having an inline controller would make sense. However I still believe the template should be in it's own html file if for no other reason than that writing HTML using JS strings is a huge pain.

**Review by Nigh7Sh4de — COMMENTED** [review]
Code is looking really good, most of my comments are nit-picky code styling things. Please let me know if you disagree with anything.

File: www/js/app.js (hunk 5 above)
Comment (iamlordvodemort) [reply www/js/app.js]: I've tried moving the controller to a seperate file, but i get errors running the app after that 
(Error: [ng:areq] Argument 'ProfileBarController' is not a function, got undefined) If you can let me know how to do this. Same with loading the template from a seperate html file.

File: www/js/settingsCtrl.js (hunk 6 above)
Comment (iamlordvodemort) [reply www/js/settingsCtrl.js]: this was left over from something i was testing, will remove it

File: www/js/app.js (hunk 5 above)
Comment (Nigh7Sh4de) [reply www/js/app.js]: So make sure that you have added the new .js file to the index.html so it gets loaded in and that you are declaring the new controller correctly (although if you already moved the directive out into a new file, then you can just declare it inline as you already have).

File: www/js/settingsCtrl.js (hunk 6 above)
Comment (Nigh7Sh4de) [reply www/js/settingsCtrl.js]: Awesome thanks :)

**Review by Nigh7Sh4de — COMMENTED** [review]
Just the one console log left and then I'll accept. Great work!

**Pull request merged**